### PR TITLE
Add `verbose` option

### PR DESCRIPTION
**Pull request by MarkTiedemann**
My first pull request. :)

- Adds a `verbose` option to improve performance (see: [https://github.com/sindresorhus/tasklist/issues/6](https://github.com/sindresorhus/tasklist/issues/6])
- Adds an `apps` option (to return only Windows Store Apps)
- Makes the `filter` option accept not only a filter array, but also a single filter string

### CODE REVIEW

**Comment by MarkTiedemann** [comment]
CI failed for `nodejs_version=0.12` with the following output:

```
const debug = require('debug')('xo');
^^^^^
SyntaxError: Use of const in strict mode.
```

Should support for Node v0.12 get dropped or should the `xo` devDependency be downgraded?

File: index.js
Comment (sindresorhus) [inline index.js]: I prefer to keep it `Array` only. I don't like mixing types with little benefit.

File: readme.md
Comment (sindresorhus) [inline readme.md]: Write the default value like done here: https://github.com/sindresorhus/boxen#borderstyle

File: readme.md
Comment (sindresorhus) [inline readme.md]: I think it's useful for people to see the properties like this with values, so maybe add another example in the `verbose` option section and show these.

**Comment by sindresorhus** [comment]
> Should support for Node v0.12 get dropped or should the xo devDependency be downgraded?

I will do it after this PR. Just ignore Node.js 0.12 failures for now.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by MarkTiedemann** [comment]
I have made the changes you requested and even fixed a bug in my code. :)

- Make test cases check all properties
- Revert making `memUsage` a `verbose` property (which it is not)
- Revert making `filter` option accept strings
- Add example to `verbose` option documentation
- Change formatting of default values in documentation

**Comment by MarkTiedemann** [comment]
Just noticed CI failed, and for good reason:
- It seems like `windowTitle` won't be set in the CI env
- The `apps` version returns different rows: 
    - Specifically, it does not have `sessionName` and `sessionNumber`, but instead it has `packageName`
    - In `verbose` mode, it has both `sessionName` and `sessionNumber` as well as `packageName`

Gonna update the PR as soon as possible. :)

**Comment by MarkTiedemann** [comment]
Earlier CI failed on Node 0.12 with linting errors, now it failed on Node 4, too.

**Comment by sindresorhus** [comment]
@MarkTiedemann What's the status of this PR? :)

**Comment by MarkTiedemann** [comment]
Well, I (almost) forgot about the PR. ;)

But thanks to your question, last night I spent 2-3 hours revisiting, upgrading, re-testing, fixing, and completely refactoring the old code. :)

Not 100% happy with the results yet, and some documentation is still missing, but I'm probably gonna push the results in the next few days.

PS: Last time I worked on this PR a couple of weeks ago, I created a streaming version of this module: [tasklist-stream](https://github.com/MarkTiedemann/tasklist-stream). I think once this PR is done, we could perhaps merge both in some way (they do "share" a lot of code already ...).

**Comment by MarkTiedemann** [comment]
@sindresorhus Here's my refactoring:

 - Update Node versions in appveyor.yml
 - Update dependencies
 - Use const instead of var
 - Use () => {} instead of function() {}
 - Remove Promise ponyfill
 - Remove 'N/A' check due to failure on non-English systems
 - Make the tests cases cover all returned tasks

Hope you like it. :)

File: test.js
Comment (sindresorhus) [inline test.js]: Can you use a [test macro](https://github.com/avajs/ava#test-macros) instead?

File: test.js
Comment (sindresorhus) [inline test.js]: Use a for-of loop.

File: test.js
Comment (sindresorhus) [inline test.js]: Don't change this.

File: package.json
Comment (sindresorhus) [inline package.json]: Don't change this.

File: package.json
Comment (sindresorhus) [inline package.json]: Don't change this.

File: package.json
Comment (sindresorhus) [inline package.json]:
```
"node": ">=4"
```

File: test.js
Comment (sindresorhus) [inline test.js]: Use async/await

File: index.js
Comment (sindresorhus) [inline index.js]: Use for-of loop.

File: index.js
Comment (sindresorhus) [inline index.js]: Don't change this. I have it like this so it will be super easy to change to default parameters when I can target Node.js 6.

**Review by sindresorhus — CHANGES_REQUESTED** [review]
Looks really good. Just some minor nitpick.

File: index.js
Comment (MarkTiedemann) [reply index.js]: There was no for-loop before. I'm gladly changing it, but may I ask why?

File: index.js
Comment (MarkTiedemann) [reply index.js]: I didn't think about that. Thanks for explaining. :)

File: test.js
Comment (MarkTiedemann) [reply test.js]: I didn't like the inconsistency of having require statements in the index file and import statements in the test file. But I guess your plan is to change to import statements everywhere in the long run, so yeah, gladly changing it back. :)

File: test.js
Comment (MarkTiedemann) [reply test.js]: Been using ava since .9, I totally missed that you support test macros now. Thanks for the advice!

File: index.js
Comment (sindresorhus) [reply index.js]: It's faster. I'm trying to use it over forEach whenever possible. So for consistency too. And I don't like inline arrow functions for things that doesn't return anything. Makes the code harder to read.

Comment (MarkTiedemann) [reply index.js]: Okay, I understand that. Thanks for explaining!

PS: I have fixed the requested changes. :)

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
Looks great. Thank you @MarkTiedemann :)

**Comment by sindresorhus** [comment]
Would you be interested in being a maintainer here? I don't think this module requires much more work, so it's mostly just to highlight your contribution here in the readme.

**Comment by MarkTiedemann** [comment]
Yeah, sure. :) Thanks a lot!

Well, there's some work I'd like to do here in the time to come:
- Implement the `/apps` flag
- Implement the `/m` module flag
- Implement the `/svc` service flag
- Make a streaming interface available (I already created a prototype at [tasklist-stream](https://github.com/MarkTiedemann/tasklist-stream), but there's more work to be done there, too. And I'm not sure how to "merge" both modules - if they should be merged at all. 
- As per `tasklist /?`, `windowTitle` and `status` filters are not supported when querying a remote machine. I'd like to implement a check for that, and also test whether this case breaks the code of this module in any way.

**Comment by sindresorhus** [comment]
Awesome! Could you put the above into an issue with checklists?

**Comment by MarkTiedemann** [comment]
Yeah, sure. See: https://github.com/sindresorhus/tasklist/issues/8.

**Pull request merged**
